src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -86,9 +86,7 @@ Repository getRepository(Long participationId, RepositoryActionType repositoryAc
         }
 
         // Add submission policy to the programming exercise.
-        if (programmingExercise.getSubmissionPolicy() == null) {
```

Comment:
The way the participation and programming exercise are retrieved above, the submission policy will always be "not initialized" (not null!).